### PR TITLE
ARROW-10748: [Java][JDBC] Support consuming timestamp data when time zone is not available

### PR DESCRIPTION
**Pull request by liyafan82**
This PR fixes a problem reported by the user. 
See the description in https://issues.apache.org/jira/browse/ARROW-10748

### CODE REVIEW

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowVectorIteratorTest.java [hunk 1]
```diff
@@ -92,6 +103,39 @@ public void testJdbcToArrowValuesNoLimit() throws SQLException, IOException {
     validate(iterator);
   }
 
+  @Test
+  public void testTimeStampConsumer() throws SQLException, IOException {
```
Comment (emkornfield) [inline java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowVectorIteratorTest.java:107]: nit: Timestamp?

Comment (emkornfield) [reply java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowVectorIteratorTest.java:107]: oh I see arrow's Vectors are inconsistent here.

**Comment by emkornfield** [comment]
+1

**Comment by liyafan82** [comment]
@emkornfield Thanks for your comments.

**Pull request closed without merging**